src/components/select-menu/useSelectMenuAnimations.tsx

```diff
@@ -0,0 +1,199 @@
+
+      popupContent.style.width = `${initialContainerSize.width}px`;
+      popupContent.style.height = `${initialContainerSize.height}px`;
+      popupContent.classList.add(SCROLL_HIDE_CLASSNAME);
+
+      if (!hasReduceMotion) {
+        // Reset the popup height to the pre-appear position
+        popupContainer.style.height = `1px`;
+        popupContainer.style.opacity = `0`;
+
+        // Popup width at the start of animation
+        // should be the same as element select width
+        popupContainer.style.width = `${selectElementSize.width}px`;
+
+        resetFloatingContainerTopPosition(floatingContainer, lastRef);
+      }
+
+      // Wait for the next frame so we
+      // know all the style changes have
+      // taken hold.
+      requestAnimationFrame(() => {
+        if (!hasReduceMotion) {
+          popupContainer.classList.add(ANIMATE_CLASSNAME);
+          floatingContainer.classList.add(ANIMATE_CLASSNAME);
+        }
+
+        floatingContainer.classList.add(OPEN_CLASSNAME);
+
+        popupContainer.style.opacity = `1`;
+
+        popupContainer.style.height = `${initialContainerSize.height}px`;
+        const popupWidth: number = Math.max(
+          initialContainerSize.width,
+          selectElementSize.width * 0.7,
+          MIN_POPUP_WIDTH
+        );
+
+        popupContainer.style.width = `${popupWidth}px`;
+        popupContent.style.width = `${popupWidth}px`;
+
+        // Animate the floating container position back to it's initial state
+        floatingContainer.style.top = `${initialContainerSize.top}px`;
+        // Ensure manipulating popup height doesn't affect the floating container
+        floatingContainer.style.height = `${initialContainerSize.height}px`;
+        floatingContainer.style.width = `${Math.max(
+          initialContainerSize.width,
+          popupWidth
+        )}px`;
+
+        const selectedElement = select.querySelector('[aria-selected=true]');
+
+        // Scroll to selected element if any
+        if (selectedElement && popupContent) {
+          const elementRect = selectedElement.getBoundingClientRect();
+          const popupRect = popupContent.getBoundingClientRect();
+          const scrollAmount =
+            elementRect.top - popupRect.top - popupRect.height / 2;
+
+          popupContent.scroll({
+            top: scrollAmount,
+          });
+        }
+
+        const cleanupAfterTransition = popupContent => {
+          popupContent.classList.remove(SCROLL_HIDE_CLASSNAME);
+        };
+
+        const handleTransitionEnd = e => {
+          // Once height finishes transition
+          // we are sure the component is fully visible
+          if (e.propertyName === 'height') {
+            cleanupAfterTransition(popupContent);
+
+            popupContainer.removeEventListener(
+              'transitionend',
+              handleTransitionEnd
+            );
+          }
+        };
+
+        popupContainer?.addEventListener('transitionend', handleTransitionEnd);
```

Comment:
why null coalescing in `popupContainer?`

Reply:
Some leftover, thanks for catching!